From 6a5e1f7a8a3f5470d781dd86e6391bd28c1dd2a2 Mon Sep 17 00:00:00 2001
From: George Zacharia <george.zcharia@gmail.com>
Date: Sun, 22 Mar 2020 05:40:48 +0530
Subject: [PATCH] APK optimisations

Drop all translations
zipAlign output
Lint shouldn't abort
Give custom name to output
---
 app/build.gradle | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/app/build.gradle b/app/build.gradle
index 356880892..4a909ed85 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -15,6 +15,11 @@ android {
         enabled = true
     }
 
+    lintOptions {
+        disable 'MissingTranslation'
+        abortOnError false
+    }
+    
     defaultConfig {
         minSdkVersion versions.minSdk
         targetSdkVersion versions.targetSdk
@@ -78,10 +83,12 @@ android {
             versionCode versions.code
             versionName versions.name
 
+            resConfigs "en"
             manifestPlaceholders = [
                     androidBackupKey: rootProject.hasProperty("SG_ANDROID_BACKUP_KEY")
                             ? rootProject.property("SG_ANDROID_BACKUP_KEY") : "MISSING"
             ]
+            archivesBaseName = "SeriesGuide-v$versionName-"+getDate()
         }
         amazon {
             applicationId "com.uwetrottmann.seriesguide.amzn"
@@ -112,6 +119,7 @@ android {
 
             multiDexEnabled false
             minifyEnabled true
+            zipAlignEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
 
             if (rootProject.file('keystore.properties').exists()) {
@@ -281,3 +289,8 @@ def propertyOrEmpty(String name) {
 def propertyOrNull(String name) {
     return rootProject.hasProperty(name) ? "\"${rootProject.property(name)}\"" : "null"
 }
+
+def static getDate() {
+    def date = new Date()
+    return date.format('ddMM')
+}
